utilities/Read-MAC-Address/Helpers.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Net.NetworkInformation;
+using System.Text;
+
+namespace Helpers
+{
+    public static class Helpers
+    {
+        /// <summary>
+        /// Return MAC Address from network interface.
+        /// </summary>
+        /// <returns>String from "First" Converted Physical Address</returns>
+        /// <remarks>Usage: string mac = Utilities.GetMacId();</remarks>
+        public static string GetMacId()
+        {
+            NetworkInterface[] nis = NetworkInterface.GetAllNetworkInterfaces();
+            if (nis.Length > 0)
+            {
+                // get the first interface
+                NetworkInterface ni = nis[0];
+                return ByteArrayToHex(ni.PhysicalAddress);
+            }
+            else
+            {
+                return "000000000000";
+            }
+        }
+
+        /// <summary>
+        /// Return String from byte[].
+        /// </summary>
+        /// <param name="barray">The value to convert. Supported types are: byte[].</param>
+        /// <returns>String from byte array</returns>
+        /// <remarks>Usage: string b = Utilities.ByteArrayToHex(new byte[] { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 });.</remarks>
+
+        public static string ByteArrayToHex(byte[] barray)
+        {
+            char[] c = new char[barray.Length * 2];
```

Comment:
There is a much easier way of doing this.
Check this code here: https://github.com/nanoframework/Samples/blob/62f77443c624143883a333f4c8a24c193d0f00e0/samples/ToStringTest/ToStringTestingShared/TestCode.cs#L54

Reply:
Yes easier, but is it unit testable ;-) 
Which method do you prefer I change it in, ByteArrayToHex or GetMacId ?

Reply:
Since you already have that method in place just use the ByteArrayToHex .